packages/position/src/react/index.tsx

```diff
@@ -0,0 +1,170 @@
+import * as React from 'react'
+import {
+  canUseDOM,
+  cloneElementWithRef,
+  createUniversalPortal
+} from '@pluralsight/ps-design-system-util'
+
+import * as positionFns from '../js'
+import useOnWindowResize from './use-on-window-resize'
+import useOnWindowScroll from './use-on-window-scroll'
+
+export const Above = React.forwardRef<
+  HTMLElement,
+  Omit<PositionProps, 'position'>
+>((props, ref) => {
+  return <Position position={positionFns.above} ref={ref} {...props} />
+})
+Above.displayName = 'Above'
+
+export const AboveLeft = React.forwardRef<
+  HTMLElement,
+  Omit<PositionProps, 'position'>
+>((props, ref) => {
+  return <Position position={positionFns.aboveLeft} ref={ref} {...props} />
+})
+AboveLeft.displayName = 'AboveLeft'
+
+export const AboveRight = React.forwardRef<
+  HTMLElement,
+  Omit<PositionProps, 'position'>
+>((props, ref) => {
+  return <Position position={positionFns.aboveRight} ref={ref} {...props} />
+})
+AboveRight.displayName = 'AboveRight'
+
+export const Below = React.forwardRef<
+  HTMLElement,
+  Omit<PositionProps, 'position'>
+>((props, ref) => {
+  return <Position position={positionFns.below} ref={ref} {...props} />
+})
+Below.displayName = 'Below'
+
+export const BelowLeft = React.forwardRef<
+  HTMLElement,
+  Omit<PositionProps, 'position'>
+>((props, ref) => {
+  return <Position position={positionFns.belowLeft} ref={ref} {...props} />
+})
+BelowLeft.displayName = 'BelowLeft'
+
+export const BelowRight = React.forwardRef<
+  HTMLElement,
+  Omit<PositionProps, 'position'>
+>((props, ref) => {
+  return <Position position={positionFns.belowRight} ref={ref} {...props} />
+})
+BelowRight.displayName = 'BelowRight'
+
+export const RightOf = React.forwardRef<
+  HTMLElement,
+  Omit<PositionProps, 'position'>
+>((props, ref) => {
+  return <Position position={positionFns.rightOf} ref={ref} {...props} />
+})
+RightOf.displayName = 'RightOf'
+
+export const LeftOf = React.forwardRef<
+  HTMLElement,
+  Omit<PositionProps, 'position'>
+>((props, ref) => {
+  return <Position position={positionFns.leftOf} ref={ref} {...props} />
+})
+LeftOf.displayName = 'LeftOf'
+
+interface ShowProps {
+  ref?: React.RefObject<HTMLElement>
+  style?: React.CSSProperties
+}
+interface PositionProps {
+  children: React.FunctionComponentElement<{}>
```

Comment:
what is `FunctionComponentElement` and is that better to use here than `ReactNode` or `ReactElement`

Reply:
Franky, I'm not 100% sure and still learning.  First timed used.  Though I think I also use it in Tab.

From the typings, it's a ReactElement with a ref:

```
    interface FunctionComponentElement<P> extends ReactElement<P, FunctionComponent<P>> {
        ref?: 'ref' extends keyof P ? P extends { ref?: infer R } ? R : never : never;
    }
```

Which is what I wanted. It's used here because it allows me to see the ref on the child and the props on the child. 

This was a major challenge in this refactor.